### PR TITLE
eth/fetcher: fix test to avoid hanging. Partial fix for #23331

### PR DESCRIPTION
**Pull request by holiman**
Issue #23331 contains some very interesting cases of hard-to-find subtle bugs. This PR contains (so far) partial fixes for them. I'm not sure how to best verify the fixes though. 

This PR fixes (?) the following ones:

- [ ] 1. 
https://github.com/ethereum/go-ethereum/blob/56e9001a1a8ddecc478943170b00207ef46109b9/eth/downloader/downloader_test.go#L1517
- [x] 2. https://github.com/ethereum/go-ethereum/blob/56e9001a1a8ddecc478943170b00207ef46109b9/eth/fetcher/tx_fetcher_test.go#L1127
- [x] 3. https://github.com/ethereum/go-ethereum/blob/d8ff53dfb8a516f47db37dbc7fd7ad18a1e8a125/eth/fetcher/tx_fetcher_test.go#L1213
- [ ] 4. https://github.com/ethereum/go-ethereum/blob/56e9001a1a8ddecc478943170b00207ef46109b9/eth/filters/filter_system_test.go#L217
- [x] 5. https://github.com/ethereum/go-ethereum/blob/56e9001a1a8ddecc478943170b00207ef46109b9/eth/filters/filter_system_test.go#L274
- [ ] 6. https://github.com/ethereum/go-ethereum/blob/56e9001a1a8ddecc478943170b00207ef46109b9/eth/filters/filter_system_test.go#L477

### CODE REVIEW

**Comment by holiman** [comment]
Rebased, another case fixed

**Comment by holiman** [comment]
I think the last one, for `TestPendingLogsSubscription` was fixed by https://github.com/ethereum/go-ethereum/pull/23619

File: eth/fetcher/tx_fetcher_test.go [hunk 1]
```diff
@@ -1263,6 +1263,17 @@ func testTransactionFetcher(t *testing.T, tt txFetcherTest) {
 	fetcher.Start()
 	defer fetcher.Stop()
 
+	defer func() { // drain the wait chan on exit
+		for {
+			select {
+			case <-wait:
+				continue
```
Comment (fjl) [inline eth/fetcher/tx_fetcher_test.go:1270]: This `continue` is not needed.

File: eth/fetcher/tx_fetcher_test.go [hunk 2]
```diff
@@ -304,7 +304,7 @@ func TestTransactionFetcherSingletonRequesting(t *testing.T) {
 func TestTransactionFetcherFailedRescheduling(t *testing.T) {
 	// Create a channel to control when tx requests can fail
 	proceed := make(chan struct{})
-
+	defer close(proceed)
```
Comment (holiman) [inline eth/fetcher/tx_fetcher_test.go:307]: TODO: look into why this is needed

Comment (karalabe) [inline eth/fetcher/tx_fetcher_test.go:307]: IMHO this is a bad change. The `proceed` channel is used to tightly control when the fetchers can tick ahead. If there's a leftover hanging, waiting to proceed, we should figure out if it's correct or not. And if correct, we should add a doFunc step that permits it to continue. If it's incorrect, we need to find why. Closing the channel just blindly unblocks stuff that might be incorrectly blocked.

**Pull request merged**